devito/finite_differences/derivative.py

```diff
@@ -213,6 +213,14 @@ def subs(self, *args, **kwargs):
         # because a new derivative could be created i.e `f.dx.subs(f.dx, y)`
         if len(args) == 2 and args[0] == self:
             return args[1]
+        # Check if we are calling subs(self, {old: new}, **hint) in which case
+        # return the standard substitution.
+        elif len(args) == 1 and self in args[0]:
+            new = args[0].pop(self)
```

Comment:
just use `self`?

Reply:
No this is `args[0][self]` not `self`. I.e this extract what to replace `self` with.